fortuna/src/chain/reader.rs

```diff
@@ -0,0 +1,89 @@
+use {
+    anyhow::Result,
+    axum::async_trait,
+    ethers::types::Address,
+};
+
+/// EntropyReader is the read-only interface of the Entropy contract.
+#[async_trait]
+pub trait EntropyReader: Send + Sync {
+    /// Get an in-flight request (if it exists)
+    /// Note that if we support additional blockchains in the future, the type of `provider` may
```

Comment:
I think you might want to deal with the generic here now, because this trait is a public interface. If we add a new chain this will break peoples working code and if It's not generic it defeats the purpose of the trait in the first place. You cant get here quite fast with a less likely to break trait using associated types:

```
pub trait EntropyReader: Send + Sync {
    type Provider;
    async fn get_request(&self, provider: Self::Provider, sequence_number: u64)
        -> Result<Option<Request>>;
}
```

Then in your ethereum.rs you can do:

```
impl EntropyReader for PythContract {
    type Provider = Address;
```

To associate the Ethereum type for that implementation.